### DISCUSSION TITLE
How do you handle device clock skews

### DISCUSSION THREAD

**Discussion by kavithachandramohan-okta** [MobileNativeFoundation/discussions#164] [General]
Clock skews or device time modified by user breaks jwt-based authentication. How do you handle this use case and any reliable NTP library recommendations?

**Comment by keith** [comment]
We wrote and still use this NTP library at Lyft https://github.com/MobileNativeFoundation/Kronos

**Reply by kavithachandramohan-okta** [reply]
We were looking into this and instacart library (https://github.com/instacart/truetime-android).  Thanks for confirming that this is still actively used and maintained 🙏

**Reply by keith** [reply]
Definitely. The android version of this library is actually still on the lyft github org https://github.com/lyft/kronos-android, but definitely file issues if you find any!
